docs/blog/announcing-vite3.md

```diff
@@ -0,0 +1,228 @@
+
+### Improved WebSocket Connection Strategy
+
+One of the pain points of Vite v2 was configuring the server when running behind a proxy. Vite v3 changes the default connection scheme so it works out of the box in most scenarios. All these setups are now tested as part of the Vite Ecosystem CI through [`vite-setup-catalogue`](https://github.com/sapphi-red/vite-setup-catalogue).
+
+### ESM SSR build by default
+
+Most SSR frameworks in the ecosystem were already using ESM builds. So, Vite v3 makes ESM the default format for SSR builds. This allows us to streamline previous [SSR externalization heuristics](https://vitejs.dev/guide/ssr.html#ssr-externals), externalizing dependencies by default.
+
+### Cold Start Improvements
+
+Vite now avoids full reload during cold start when imports are injected by plugins while crawling the initial statically imported modules ([#8869](https://github.com/vitejs/vite/issues/8869)).
+
+In Vite 2.9, both the scanner and optimizer were run in the background. In the best scenario, where the scanner would find every dependency, no reload was needed in cold start. But if the scanner missed a dependency, a new optimization phase and then a reload were needed. Vite was able to avoid some of these reloads in v2.9, as we detected if the new optimized chunks were compatible with the ones the browser had. But if there was a common dep, the sub-chunks could change and a reload was required to avoid duplicated state. In Vite 3, the optimized deps aren't handed to the browser until the crawling of static imports is done. A quick optimization phase is issued if there is a missing dep (for example, injected by a plugin), and only then, the bundled deps are sent. So, a page reload is no longer needed for these cases.
+
+<img style="background-color: var(--vp-code-block-bg);padding:4%;border-radius:8px;" width="100%" height="auto" src="../images/vite-3-cold-start.svg">
+
+### import.meta.glob
+
+`import.meta.glob` support was rewritten. Read about the new features in the [Glob Import Guide](https://main.vitejs.dev/guide/features.html#glob-import):
+
+[Multiple Patterns](https://vitejs.dev/guide/features.html#multiple-patterns) can be passed as an array
+
+```js
+import.meta.glob(['./dir/*.js', './another/*.js'])
+```
+
+[Negative Patterns](https://vitejs.dev/guide/features.html#negative-patterns) are now supported (prefixed with `!`) to ignore some specific files
+
+```js
+import.meta.glob(['./dir/*.js', '!**/bar.js'])
+```
+
+[Named Imports](https://vitejs.dev/guide/features.html#named-imports) can be specified to improve tree-shacking
+
+```js
+import.meta.glob('./dir/*.js', { import: 'setup' })
+```
+
+[Custom Queries](https://vitejs.dev/guide/features.html#custom-queries) can be passed to attach metadata
+
+```js
+import.meta.glob('./dir/*.js', { query: { custom: 'data' } })
+```
+
+[Eager Imports](https://vitejs.dev/guide/features.html#glob-import) is now passed as a flag
```

Comment:
I think sapphi suggested that we remove the domain name. I updated them below but GitHub panicked, so maybe it's easier to fix by editing my comment and copy-paste the changes.

```suggestion
`import.meta.glob` support was rewritten. Read about the new features in the [Glob Import Guide](/guide/features.html#glob-import):

[Multiple Patterns](/guide/features.html#multiple-patterns) can be passed as an array

```js
import.meta.glob(['./dir/*.js', './another/*.js'])
```

[Negative Patterns](/guide/features.html#negative-patterns) are now supported (prefixed with `!`) to ignore some specific files

```js
import.meta.glob(['./dir/*.js', '!**/bar.js'])
```

[Named Imports](/guide/features.html#named-imports) can be specified to improve tree-shacking

```js
import.meta.glob('./dir/*.js', { import: 'setup' })
```

[Custom Queries](/guide/features.html#custom-queries) can be passed to attach metadata

```js
import.meta.glob('./dir/*.js', { query: { custom: 'data' } })
```

[Eager Imports](/guide/features.html#glob-import) is now passed as a flag
```

Reply:
done